orchestra/models.py

```diff
@@ -461,6 +461,43 @@ def __str__(self):
             str(self.task), self.assignment_counter, str(self.worker))
 
 
+class Iteration(models.Model):
+    """
+    Iterations are the contiguous units of a worker's time on task.
+
+    Attributes:
+        start_datetime (datetime.datetime):
+            The time the task was assigned for this iteration.
+        end_datetime (datetime.datetime):
+            The time the iteration was completed. Will be None if the iteration
+            is currently in progress.
+        assignment (orchestra.models.TaskAssigment):
+            The task assignment to which the iteration belongs.
+        action (orchestra.models.Iteration.Action):
```

Comment:
`REJECTED` sounds like the iteration itself was rejected, when in reality it just means that the worker is passing it back down the hierarchy. I do like the idea of having processing and invalidated statuses, though.